ooyala_player/public/js/ooyala_player.js

```diff
@@ -49,6 +54,35 @@ function OoyalaPlayerBlock(runtime, element) {
             // ToDo: (APROS Specific) Remove after fixing APROS JS
             $('#course-lessons').off('mouseup');
         },
+        updateCompletion: function(context){
+            var playing = context.player.isPlaying();
+            var percentComplete = context.player.getPlayheadTime() / context.player.getDuration();
+            if (playing === true){
```

Comment:
@macornwell I'm not able to understand the reason behind this check and setting timeout. Could you please explain?

Reply:
@ziafazal in order to get when the video is coming near completion, it needs to be examined periodically. The timeout is to check the current time of the video and determine if it is close to completion.